core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -5053,36 +5053,32 @@ public void setPreparedStatementValue( DatabaseMeta databaseMeta, PreparedStatem
           }
           break;
         case ValueMetaInterface.TYPE_STRING:
-          if ( getLength() < databaseMeta.getMaxTextFieldLength() ) {
-            if ( !isNull( data ) ) {
-              preparedStatement.setString( index, getString( data ) );
-            } else {
-              preparedStatement.setNull( index, java.sql.Types.VARCHAR );
+          if ( !isNull( data ) ) {
+            if ( getLength() == DatabaseMeta.CLOB_LENGTH ) {
+              setLength( databaseMeta.getMaxTextFieldLength() );
             }
-          } else {
-            if ( !isNull( data ) ) {
-              String string = getString( data );
-
-              int maxlen = databaseMeta.getMaxTextFieldLength();
-              int len = string.length();
-
+            String string = getString( data );
+            int len = string.length();
+            int maxlen = isLengthInvalidOrZero() ? 0 : getLength();
```

Comment:
@cravobranco It's here.

Reply:
@pentaho-lmartins you are correct, it was my mistake, sorry.